src/erlcloud_application_autoscaler.erl

```diff
@@ -0,0 +1,689 @@
+    BodyConfigurations :: aws_aas_request_body()
+) -> response().
+register_scalable_target(Configurations, BodyConfigurations) ->
+    request_with_action(Configurations, BodyConfigurations, "AnyScaleFrontendService.RegisterScalableTarget").
+
+
+%%%------------------------------------------------------------------------------
+%%% Internal functions.
+%%%------------------------------------------------------------------------------
+
+request_with_action(Configuration, BodyConfigurations, Action) ->
+    Body = jsx:encode(BodyConfigurations),
+    case erlcloud_aws:update_config(Configuration) of
+        {ok, Config} ->
+            HeadersPrev = headers(Config, Action, Body),
+            Headers = [{"content-type", "application/x-amz-json-1.1"} | HeadersPrev],
+            request_and_retry(Config, Headers, Body, {attempt, 0});
+        {error, Reason} ->
+            {error, Reason}
+    end.
+
+request_and_retry(_, _, _, {error, Reason}) ->
+    {error, Reason};
+request_and_retry(_, _, _, {attempt, Attempt}) when Attempt =:= 11 ->
+    {error, no_answer};
+request_and_retry(Config, Headers, Body, {attempt, Attempt}) ->
+    case erlcloud_httpc:request(url(Config), post, Headers, Body, timeout(Attempt, Config), Config) of
+
+        {ok, {{200, _}, _, <<>>}} ->
+            ok;
+
+        {ok, {{200, _}, _, RespBody}} ->
+            {ok, jsx:decode(RespBody)};
+
+        {error, {_, timeout}} ->
+            request_and_retry(Config, Headers, Body, retry({attempt, Attempt + 1}));
+        Error ->
+            request_and_retry(Config, Headers, Body, retry(Error))
+    end.
+
+retry({attempt, Attempt}) when Attempt >= ?NUM_ATTEMPTS ->
+    {error, max_limit_of_attempts};
+retry({attempt, Attempt}) ->
+    backoff(Attempt),
+    {attempt, Attempt + 1}.
+
+%% Sleep after an attempt
+-spec backoff(pos_integer()) -> ok.
+backoff(1) -> ok;
+backoff(Attempt) ->
+    timer:sleep(erlcloud_util:rand_uniform((1 bsl (Attempt - 1)) * 100)).
+
+-spec timeout(pos_integer(), aws_config()) -> pos_integer().
+%% HTTPC timeout for a request
+timeout(1, #aws_config{timeout = undefined}) ->
+    %% Shorter timeout on first request. This is to avoid long (5s) failover when first DDB
+    %% endpoint doesn't respond
+    1000;
+timeout(_, #aws_config{} = Cfg) ->
+    erlcloud_aws:get_timeout(Cfg).
+
+
+-spec url(aws_config()) -> [].
+url(#aws_config{autoscaling_scheme = Scheme, autoscaling_host = Host} = Config) ->
+    lists:flatten([Scheme, Host, port_spec(Config)]).
+
+port_spec(#aws_config{autoscaling_port=80}) ->
+    "";
+port_spec(#aws_config{autoscaling_port=Port}) ->
+    [":", erlang:integer_to_list(Port)].
+
+
+headers(Config, Operation, Body) ->
+    Headers = [{"host", Config#aws_config.autoscaling_host},
+               {"x-amz-target", Operation}],
+
+    erlcloud_aws:sign_v4_headers(Config, Headers, Body, erlcloud_aws:aws_region_from_host(Config#aws_config.autoscaling_host), "application-autoscaling").
```

Comment:
not sure if reuse of autoscaling_host inside the record is a good idea. suggest to have a dedicated one.

Reply:
I am a bit confused about this comment, could you elaborate a bit more, or give an example of how it could go wrong? Is it because it can be confused with the `as_host`, or is there something else I'm missing ?

Reply:
Or can it be that I may have been mistaken and that this endpoint is actually the EC2/basic autoscaling, as is the `as_host`, and it would be more correct to have something like `aws_config.application_autoscaling_*` fields?

Reply:
sorry confusing comment. yes, the latter is what i meant. please use a dedicate host config params. Smth like `app_autoscalling`